linera-core/src/local_node.rs

```diff
@@ -462,7 +463,7 @@ where
         chain_id: ChainId,
         mut start: BlockHeight,
         stop: BlockHeight,
-        notifications: &mut impl Extend<Notification>,
+        notifications: &impl NotificationSink<Notification>,
```

Comment:
nit: If we're going to define our own trait, probably `notifier: impl Notifier` (no generic) then `notifier.notify(&notifications)` would be cleaner.

Reply:
I had to rename the `struct Notifier` to avoid the name collision. I called it `ChannelNotifier` for now.